src/mixins/navigator/filteredChildren.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2023-2024 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import { TOPIC_TYPE_TO_TAG, HIDE_DEPRECATED } from 'docc-render/constants/Tags';
+import { ChangeNames } from 'docc-render/constants/Changes';
+
+export default {
+  computed: {
+    /**
+       * Returns a list of the child nodes, that match the filter pattern.
+       * @returns NavigatorFlatItem[]
+       */
+    filteredChildren: ({
+      children,
+      selectedTags,
+      apiChanges,
+      filterPattern,
+      filterChildren,
+    }) => (filterChildren(children, selectedTags, apiChanges, filterPattern)),
+  },
+  methods: {
+    filterChildren(children, selectedTags, apiChanges, filterPattern) {
+      const tagsSet = new Set(selectedTags);
+      // find children that match current filters
+      return children.filter(({
+        title, path, type, deprecated, deprecatedChildrenCount, childUIDs,
+      }) => {
+        // groupMarkers know how many children they have and how many are deprecated
+        const isDeprecated = deprecated || deprecatedChildrenCount === childUIDs.length;
+        // check if `title` matches the pattern, if provided
+        const titleMatch = filterPattern ? filterPattern.test(title) : true;
+        // check if `type` matches any of the selected tags
+        let tagMatch = true;
```

Comment:
Could we simplify this by calling `extractTags`, and check if there's any overlap between `selectedTags`?

Reply:
`extractTags` would just give us an array of `availableTags` that we end up using as suggested tags for the filter. In the `filterChildren` method, we filter the symbols. Could you develop how `extractTags` would help us to simply this function? we always have selected tags.

Reply:
I tried it out today. Initially I thought that we could just check if there are overlaps between the `availableTags` of a given node and the `selectedTags` to know if a node should be included in the filter or not. (In this world, I thought we can just add a new function to curate that list of `availableTags` for a node. Both the filtering logic and the tags provider logic can call this helper function)
But after trying that logic I realized that I failed to consider that when the "hide deprecated" tag is available and also selected, that node should be hidden. So looking at whether any overlaps exist is not gonna work.